feat: update challenge and prototype (#7)
AYLIEN/frontend-challenge: README.md

# Frontend Challenge

> The purpose of this challenge is to evaluate your knowledge in the domain of front-end development. We were hoping you could build an Entities search application with JavaScript frameworks and libraries using AYLIEN News API. Besides, we would like you to write some tests. We are looking for a readable, easy-to-understand code that follows best practices and design patterns.

Reviewer: do we really hope? we rather expect/want it from the candidate :)

## The challenge

> We need you to develop a Client (UI) architecture that allowing the user to search among entities and see related stories at the bottom of the search bar.

Reviewer: that allowing -> that allows

Reviewer 2: ```suggestion
You are tasked with developing a Client (UI) architecture that will allow the user to search among entities and display stories related to these entities at the bottom of the search bar.
```

### Terminology
- An entity is a thing with a distinct and independent existence.
- A story contains information about a news article enriched with NLP metadata.

### Description

> For this project, you will need to provide a search bar that helps users search via entities using the `/autocomplete` endpoint and display related stories using the `/stories` endpoint.

Reviewer: you will need to provide -> you will provide

Reviewer: that helps users search -> that allows users to search

> We :heart: React and created a boilerplate project using [Create React App](https://create-react-app.dev/). However, you are free to use any frameworks and libraries you choose. The prototype is just a suggestion — you have complete creative freedom!

Reviewer: and created a -> and we already created a

Reviewer 2: ```suggestion
We :heart: React and have already created a boilerplate project using [Create React App](https://create-react-app.dev/). However, you are free to use any frameworks and libraries you choose. The prototype is just a suggestion — you have complete creative freedom!
```

<p align="center">
  <img src="https://github.com/AYLIEN/frontend-challenge/blob/master/prototype.png?raw=true">
</p>

## API

> For using API, Please register for the News API and follow the following documentations:

Reviewer: For using API, Please register -> In order to access API you have to register

Registration form for the News API service: https://newsapi.aylien.com/signup

Documentation of the autocompletes endpoint: https://docs.aylien.com/newsapi/endpoints/#autocompletes

Documentation of the stories endpoint: https://docs.aylien.com/newsapi/endpoints/#stories

Calling the API:
https://docs.aylien.com/newsapi/#calling-the-api

Using our Swagger definition with Postman (optional): https://docs.aylien.com/newsapi/#testing-the-news-api-with-postman

## Notes

> We would like you to identify the required tasks and provide a time estimation for these tasks before you start working on them. If you can split these tasks into smaller items, please also share those. It’s up to you how much time you invest, but keep in mind that we expect a meaningful contribution and also that it should be possible to review your solution in a reasonable amount of time, so please provide instructions on how to get up and running with your solution and test it. Your solution is expected as a zip file, but please also include your .git directory — we are interested in your development steps, not just the end result.

Reviewer 2: ```suggestion
Before starting the challenge, we would also like you to identify the required steps and provide a time estimate for these. If you can split these tasks into smaller items, please also share these. It’s up to you how much time you invest, but keep in mind that we expect a meaningful contribution and also that it should be possible to review your solution in a reasonable amount of time, so please provide instructions on how to get up and running with your solution and test it. Your solution is expected as a zip file, but please also include your .git directory — we are interested in your development steps, not just the end result.
```
